CIP-????/README.md

```diff
@@ -0,0 +1,98 @@
+
+```JSON
+{
+    "url": "http://example.com/signup",
+    "action": "Sign up",
+    "timestamp": 1673261248,
+    ["email": "email@example.com"] <- Optional
+}
+```
+
+### Wallet specification
+
+The wallets can improve the overall security implementing the folliwing guidelines. We RECOMMEND to show in a structured way the payload information for shake of clarity. This information should be well understanded by the users before the payload is singed.  
+
+The ```url``` field provides information about the hostname of the application. This hostname MUST be included in the wallet allow list. If a known domain A tries to sign a payload for an unknown domain B, you will be prompted with permission popup making more obvious the cross-domain interaction. When possible, the wallet SHOULD warn the user if a payload is for a different domain.
+
+The wallet SHOULD update the ```timestamp``` field to the current time just before the signature. This field ideally should match the moment just before the signature, thereby the server recieves the more updated payload possible. 
+
+### Server processing
+
+The server has ultimate responsibility of processing correctly the requests. We use the content to validate the payload. The request will be processed with the following steps:
+
+1. The server MUST check the action and the endpoint included in the request. Each route to an endpoint MUST have an associated action and a URL. The first step is to check that they match with the parameterized action.
+
+2. The server MUST check the expiration of the payload. The expiration SHOULD be enough to give time to the user to introduce the wallet password but it SHOULD NOT be too long, we RECOMMEND not more than 5 minutes.
+
+3. The server MUST validate the COSESign1 signature and check that the address inside the protected map of the signature corresponds to the public key in the COSEKey. 
+
+Additionally the server COULD extract the payload content and pass it through the server logic.
+
+## Rationale: how does this CIP achieve its goals?
+
+This specification provides the general guidelines and necessary recommendations for performing secure authenticated web3 requests in the Cardano ecosystem. It covers the two main desired characteristics for a secure payload: It must expire and it must be non-static.
+
+Another important aspect for security is how wallets process the payload. They can improve the security using the data inside the payload to warn the users about possible malicious interactions. This specification emphasizes the importance of informing users clearly about the purpose of the payloads and how wallets can use the URL field to apply allow-lists and/or cross-domain policies.
+
+Finally, it establishes the requirements and recommendations for server side processing. The server must also ensure the validity of the signature and the payload, as well as of its purpose in order to accomplish the authentication. 
```

Comment:
Would a user have to sign a transaction for every http request or just like JWT there would be a way to cache and store token somewhere and use this until it expires. 

If this is each time then it could be very annoying, usually a webapp makes many requests to the server side, even for one page.

Reply:
Exactly. This specification isn't for each request. You can use it only for the important requests like Login, Register, etc... Afterwards a JWT token can take control of the session.

I developed an example using this methodology [cardano-express-web3-skeleton)](https://github.com/jmagan/cardano-express-web3-skeleton)

Reply:
Is good for you if I mark this as resolve?